lib/array.rb

```diff
@@ -0,0 +1,15 @@
+class Array
+  def my_inject starting = nil, symbol = nil
+    starting.nil? ? result = self[0] : result = starting
+    if starting.is_a? Symbol
+      result = self[0]
+      each { |num| result = result.send starting, num }
+    elsif symbol.is_a? Symbol
+      each { |num| result = result.send symbol, num }
+    else
+      each { |num| result = yield result, num }
+    end
+    result
+  end
+end
```

Comment:
Great stuff: I see the problem. Once you use `result = self[0]` to start with the first value, you need to remove it from being added afterwards. In your code:

```ruby
result = self[0] # result == 1
each { |num| result = result.send starting, num } # will operate on self[0] again
```